src/Cech_complex/include/gudhi/Cech_complex_blocker.h

```diff
@@ -30,33 +31,109 @@ namespace cech_complex {
  * \details
  * Čech blocker is an oracle constructed from a Cech_complex and a simplicial complex.
  *
- * \tparam SimplicialComplexForProximityGraph furnishes `Simplex_handle` and `Filtration_value` type definition,
+ * \tparam SimplicialComplexForCech furnishes `Simplex_handle` and `Filtration_value` type definition,
  * `simplex_vertex_range(Simplex_handle sh)`and `assign_filtration(Simplex_handle sh, Filtration_value filt)` methods.
  *
- * \tparam Chech_complex is required by the blocker.
+ * \tparam Cech_complex is required by the blocker.
+ *
+ * \tparam Kernel CGAL kernel.
  */
-template <typename SimplicialComplexForCech, typename Cech_complex>
+template <typename SimplicialComplexForCech, typename Cech_complex, typename Kernel>
 class Cech_blocker {
+
+ public:
+
+  using Point_d = typename Kernel::Point_d;
+  // Numeric type of coordinates in the kernel
+  using FT = typename Kernel::FT;
+  // Sphere is a pair of point and squared radius.
+  using Sphere = typename std::pair<Point_d, FT>;
+
  private:
-  using Point_cloud = typename Cech_complex::Point_cloud;
 
   using Simplex_handle = typename SimplicialComplexForCech::Simplex_handle;
   using Filtration_value = typename SimplicialComplexForCech::Filtration_value;
 
+  template<class PointIterator>
+  Sphere get_sphere(PointIterator begin, PointIterator end) const {
+    Point_d c = kernel_.construct_circumcenter_d_object()(begin, end);
+    FT r = kernel_.squared_distance_d_object()(c, *begin);
+    return std::make_pair(std::move(c), std::move(r));
+  }
+
  public:
+
   /** \internal \brief Čech complex blocker operator() - the oracle - assigns the filtration value from the simplex
    * radius and returns if the simplex expansion must be blocked.
    *  \param[in] sh The Simplex_handle.
    *  \return true if the simplex radius is greater than the Cech_complex max_radius*/
   bool operator()(Simplex_handle sh) {
-    Point_cloud points;
-    for (auto vertex : sc_ptr_->simplex_vertex_range(sh)) {
-      points.push_back(cc_ptr_->get_point(vertex));
-#ifdef DEBUG_TRACES
-      std::clog << "#(" << vertex << ")#";
-#endif  // DEBUG_TRACES
+    using Point_cloud =  std::vector<Point_d>;
+    CGAL::NT_converter<FT, double> cast_to_double;
+    Filtration_value radius = 0.;
+
+    // for each face of simplex sh, test outsider point is indeed inside enclosing ball, if yes, take it and exit loop, otherwise, new sphere is circumsphere of all vertices
+    Sphere min_enclos_ball;
+    CGAL::NT_converter<double, FT> cast_to_FT;
+    min_enclos_ball.second = cast_to_FT(std::numeric_limits<double>::max());
```

Comment:
I am not fond of this. It makes sense and works for double (although building on purpose a large FT is a bit wasteful), but we should also be able to build a Cech complex with an exact unbounded Filtration_value type.
One possibility could be to use `boost::optional<Sphere const&>`, where not-set means we haven't seen a valid sphere yet, and I am using a reference because all the spheres are stored in the cache somewhere. Or equivalently a `Sphere const*` with value 0 for no-sphere. Or something else, that's just what comes to my mind, but maybe the reference/pointer thing causes too much trouble and it is easier to use `optional<Sphere>` or `Sphere` plus a `bool`.